### PR TITLE
🔦 Feature: Incomplete translation display

### PR DESCRIPTION
**Pull request by gitstart**
## Change Summary
closes: https://github.com/nocodb/nocodb/issues/6338

Complete translation display
## Change type

- [x] feat: (complete translation display)

## Test/ Verification

Provide summary of changes.
https://www.loom.com/share/c4ea0eeda6514da2838a74d0b6448ed3

## Additional information / screenshots (optional)

---
This code was written and reviewed by GitStart Community. Growing great engineers, one PR at a time.

### CODE REVIEW

File: packages/nc-gui/utils/filterUtils.ts [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { UITypes, isNumericCol, numericUITypes } from 'nocodb-sdk'
+import { getI18n } from '../plugins/a.i18n'
+
+const getTranslation = (value: string) => getI18n().global.t(value)
```
Comment (wingkwong) [inline packages/nc-gui/utils/filterUtils.ts:4]: prefer just using `t` to keep the style consistent across the code base.

```
const { t } = getI18n().global
...
return t('filterOperation.isEqual')
```

File: packages/nc-gui/components/dashboard/Sidebar/UserInfo.vue [hunk 2]
```diff
@@ -84,22 +84,20 @@ onMounted(() => {
...
-          <template v-if="!isMobileMode">
-            <NcDivider />
-            <a href="https://docs.nocodb.com" target="_blank" class="!underline-transparent">
-              <NcMenuItem>
-                <GeneralIcon icon="help" class="menu-icon mt-0.5" />
-                <span class="menu-btn"> Help Center </span>
-              </NcMenuItem>
-            </a>
-          </template>
+          <NcDivider />
```
Comment (wingkwong) [inline packages/nc-gui/components/dashboard/Sidebar/UserInfo.vue:89]: why `<template v-if="!isMobileMode">` is removed?

File: packages/nc-gui/components/dashboard/Sidebar/UserInfo.vue [hunk 3]
```diff
@@ -134,18 +132,16 @@ onMounted(() => {
...
-              <GeneralIcon v-if="isAuthTokenCopied" icon="check" class="group-hover:text-black menu-icon" />
-              <GeneralIcon v-else icon="copy" class="menu-icon" />
-              <template v-if="isAuthTokenCopied"> Copied Auth Token </template>
-              <template v-else> Copy Auth Token </template>
-            </NcMenuItem>
-            <nuxt-link v-e="['c:navbar:user:email']" class="!no-underline" to="/account/profile">
-              <NcMenuItem> <GeneralIcon icon="settings" class="menu-icon" /> Account Settings </NcMenuItem>
-            </nuxt-link>
-          </template>
+          <NcDivider />
```
Comment (wingkwong) [inline packages/nc-gui/components/dashboard/Sidebar/UserInfo.vue:135]: why `<template v-if="!isMobileMode">` is removed?

File: packages/nc-gui/components/smartsheet/column/EditOrAdd.vue [hunk 4]
```diff
@@ -84,13 +84,16 @@ const showDeprecated = ref(false)
 
 const uiTypesOptions = computed<typeof uiTypes>(() => {
   return [
-    ...uiTypes.filter(
-      (t) => geoDataToggleCondition(t) && (!isEdit.value || !t.virtual) && (!t.deprecated || showDeprecated.value),
-    ),
+    ...uiTypes
+      .filter((t) => geoDataToggleCondition(t) && (!isEdit.value || !t.virtual) && (!t.deprecated || showDeprecated.value))
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/column/EditOrAdd.vue:88]: would suggest not to use `t` here. it's look a bit confusing with `t` from translation.

File: packages/nc-gui/utils/filterUtils.ts [hunk 5]
```diff
@@ -23,21 +26,21 @@ const getNeqText = (fieldUiType: UITypes) => {
...
-  return 'is like'
+  return getTranslation('filterOperation.isLike')
 }
 
 const getNotLikeText = (fieldUiType: UITypes) => {
   if (fieldUiType === UITypes.Attachment) {
     return "filenames doesn't contain"
   }
-  return 'is not like'
+  return getTranslation('filterOperation.isNot like')
```
Comment (wingkwong) [inline packages/nc-gui/utils/filterUtils.ts:43]: unexpected space

File: packages/nc-gui/utils/filterUtils.ts [hunk 6]
```diff
@@ -1,9 +1,8 @@
 import { UITypes, isNumericCol, numericUITypes } from 'nocodb-sdk'
 import { getI18n } from '../plugins/a.i18n'
 
-const getTranslation = (value: string) => getI18n().global.t(value)
-
 const getEqText = (fieldUiType: UITypes) => {
+  const { t } = getI18n().global
```
Comment (wingkwong) [inline packages/nc-gui/utils/filterUtils.ts:5]: you may just move it outside.

**Review by wingkwong — COMMENTED** [review]
please resolve the conflicts. after all is done, remember to click re-request review.

File: packages/nc-gui/utils/filterUtils.ts (hunk 6 above)
Comment (gitstart) [reply packages/nc-gui/utils/filterUtils.ts:5]: This causes the app to crash, so I just followed same convention in `packages/nc-gui/utils/validation.ts`

Screenshot 2023-10-02 at 12 52 45

**Comment by wingkwong** [comment]
@gitstart can you resolve the conflict again?

**Review by wingkwong — CHANGES_REQUESTED** [review]
It shouldn't have 1074 files changed :D Please only include those related files.

File: packages/nc-gui/components/smartsheet/column/EditOrAdd.vue [hunk 7]
```diff
@@ -87,12 +87,17 @@ const showDeprecated = ref(false)
 const uiTypesOptions = computed<typeof uiTypes>(() => {
   return [
     ...uiTypes
-      .filter((t) => geoDataToggleCondition(t) && (!isEdit.value || !t.virtual) && (!t.deprecated || showDeprecated.value))
-      .filter((t) => !(t.name === UITypes.SpecificDBType && isXcdbBase(meta.value?.source_id))),
+      .filter(
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/column/EditOrAdd.vue:90]: why removing `.filter((t) => !(t.name === UITypes.SpecificDBType && isXcdbBase(meta.value?.source_id))),` ?

**Comment by dstala** [comment]
reopen this thread after addressing comments from @wingkwong and on the latest branch

**Pull request closed without merging**